build.gradle

```diff
@@ -164,202 +162,192 @@ subprojects {
-            systemProperty 'log4j2.contextSelector', 'org.apache.logging.log4j.core.selector.BasicContextSelector'
-
-            // Register BucketTestExtension for all tests to enable test striping
-            // Test striping allows distributing test execution across multiple runs
-            // Usage: ./gradlew test -Dtest.striping.total=N -Dtest.striping.index=I
-            //   where N is the total number of buckets (e.g., 3)
-            //   and I is the current bucket index (0 to N-1)
-            // Example: To run 1/3 of tests in first bucket:
-            //   ./gradlew test -Dtest.striping.total=3 -Dtest.striping.index=0
-            def total = System.getProperty("test.striping.total")
-            def index = System.getProperty("test.striping.index")
-
-            if (total != null) {
-                systemProperty "test.striping.total", total
-            }
-            if (index != null) {
-                systemProperty "test.striping.index", index
-            }
+        tasks.withType(Test) {
+            // Getting javadoc to compile is part of the test suite to ensure we are able to publish our artifacts
+            dependsOn project.javadoc
+        }
+        if (!sourceSets.test.allSource.files.isEmpty()) {
+            tasks.withType(Test) {
+                testLogging {
+                    events "passed", "skipped", "failed"
+                    exceptionFormat "full"
+                    showExceptions true
+                    showCauses true
+                    showStackTraces true
+                }
+                maxParallelForks = gradle.startParameter.maxWorkerCount
+
+                // Provide way to exclude particular tests from CLI
+                // e.g. ../gradlew test -PexcludeTests=**/KafkaProtobufConsumerLongTermTest*
+                if (project.hasProperty('excludeTests')) {
+                    exclude project.property('excludeTests')
+                }
 
-            systemProperty 'junit.jupiter.extensions.autodetection.enabled', true
-            // Verify assertions in tests
-            jvmArgs += ['-ea', '-XX:+HeapDumpOnOutOfMemoryError', '-Xmx2g']
-            systemProperty "junit.jupiter.execution.timeout.default", "5m"
-            jacoco {
-                enabled = true
-                def stripingSuffix = (total != null && index != null) ? "-${index}-of-${total}" : ""
-                destinationFile = layout.buildDirectory.file("jacoco/${project.name}-${name}${stripingSuffix}.exec").get().asFile
+                useJUnitPlatform()
+                //  Disable parallel test execution, see MIGRATIONS-1666
+                systemProperty 'junit.jupiter.execution.parallel.enabled', 'false'
+                systemProperty 'log4j2.contextSelector', 'org.apache.logging.log4j.core.selector.BasicContextSelector'
+                // Verify assertions in tests
+                jvmArgs = ['-ea', '-XX:+HeapDumpOnOutOfMemoryError']
```

Comment:
Curious why the parameter is being dropped? It was added to increase stability and prevent OOMs